packages/db/src/abstractRepository.ts

```diff
@@ -23,11 +25,22 @@ export abstract class Repository<I extends Id, T> {
 
   protected type: Type<T>;
 
-  protected constructor(config: IBeaconConfig, db: IDatabaseController<Buffer, Buffer>, bucket: Bucket, type: Type<T>) {
+  protected dbReadsMetrics?: ReturnType<DbMetricCounter["labels"]>;
+  protected dbWriteMetrics?: ReturnType<DbMetricCounter["labels"]>;
+
+  protected constructor(
+    config: IBeaconConfig,
+    db: IDatabaseController<Buffer, Buffer>,
+    bucket: Bucket,
+    type: Type<T>,
+    metrics?: IDbMetrics
+  ) {
     this.config = config;
     this.db = db;
     this.bucket = bucket;
     this.type = type;
+    this.dbReadsMetrics = metrics?.dbReads.labels({bucket: getBucketNameByValue(bucket) ?? ""});
```

Comment:
Should you compute getBucketNameByValue() only once and re-use?

Reply:
Yes, since every repository is separate instance of abastract repo, it's safe to compute bucket name only once in constructor if that was your question.